### PR TITLE
feat: new syntax features

### PR DESCRIPTION
**Pull request by garfr**
* Now optional newlines
* Less verbose function headers
* Basic documentation for syntax

### CODE REVIEW

File: src/lexer.c [hunk 1]
```diff
@@ -62,12 +63,30 @@ peek_c() {
...
+         lex.prev == TOK_TRUE || lex.prev == TOK_U8 || lex.prev == TOK_I8 ||
+         lex.prev == TOK_U16 || lex.prev == TOK_I16 || lex.prev == TOK_U32 ||
+         lex.prev == TOK_I32 || lex.prev == TOK_U64 || lex.prev == TOK_I64 ||
+         lex.prev == TOK_BOOL;
+}
+
+static int
 skip_whitespace() {
+  int c;
   while (!is_eof() && is_whitespace(peek_c())) {
```
Comment (wooosh) [inline src/lexer.c:78]: The lexer's implementation of is_whitespace does not handle carriage returns as whitespace. I'd use `isspace` from ctype.h here instead.

Comment (wooosh) [reply src/lexer.c:78]: Actually, I'm not sure this is a good idea given the rest of the lexer does not support carriage returns, which is fine IMO.

**Review by wooosh — APPROVED** [review]

File: docs/language.md [hunk 2]
```diff
@@ -0,0 +1,87 @@
+# Beans Language
+
+A beans program is made up of a series of functions, the order of these functions does not matter and they can be references before being called.
```
Comment (wooosh) [inline docs/language.md]: s/references/referenced

Comment (wooosh) [inline docs/language.md]:
```suggestion
A beans program is made up of a series of functions, the order of these functions does not matter and they can be referenced before being called.
```

File: docs/language.md [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+main() {}
+```
+
+## Statements
+
+A statement makes up functions, but beans has most language features as expressions so they can be composed in powerful ways. However there are still a couple features that are only represented as statements.
+
+* Let        - introduces a new name into the current local scope, optionally giving it an initial value
+* Return     - returns from the current function, optionally with a value
+* Expression - just an expression which is computed, and the result is discarded
```
Comment (wooosh) [inline docs/language.md:43]: Are expressions that can be proved to have no side effects valid?

**Review by wooosh — CHANGES_REQUESTED** [review]

File: docs/language.md (hunk 2 above)
Comment (garfr) [reply docs/language.md]: https://github.com/garfr/bcc2/pull/3/commits/d48ea66874e5a41f7bd86ec8c6b141143ad3a86a

File: docs/language.md (hunk 3 above)
Comment (garfr) [reply docs/language.md:43]: Currently yes, and I don't see a reason to disallow pure expressions specifically, though a case can be made for not allowing expressions that evaluate to a non-unit value.

File: src/lexer.c (hunk 1 above)
Comment (garfr) [reply src/lexer.c:78]: yeah ignoring carriage returns was the previous choice

**Pull request merged**